src/main/java/com/example/bookshop/service/implementation/OrderServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.example.bookshop.service.implementation;
+
+import com.example.bookshop.dto.order.OrderDto;
+import com.example.bookshop.dto.order.OrderItemDto;
+import com.example.bookshop.dto.order.ShippingAddressRequestDto;
+import com.example.bookshop.dto.order.UpdateStatusDto;
+import com.example.bookshop.exception.EntityNotFoundException;
+import com.example.bookshop.mapper.OrderItemMapper;
+import com.example.bookshop.mapper.OrderMapper;
+import com.example.bookshop.model.Order;
+import com.example.bookshop.model.User;
+import com.example.bookshop.repository.OrderRepository;
+import com.example.bookshop.service.OrderService;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class OrderServiceImpl implements OrderService {
+    private OrderRepository orderRepository;
+    private OrderMapper orderMapper;
+    private OrderItemMapper orderItemMapper;
+
+    @Override
+    public Page<OrderDto> findAll(Pageable pageable) {
+        return new PageImpl<>(orderRepository.findAll(pageable)
+                .stream()
+                .map(orderMapper::toDto)
+                .toList());
+    }
+
+    @Override
+    public OrderDto save(ShippingAddressRequestDto shippingAddressRequestDto,
+                                User user) {
+        Order order = orderRepository.findByUser_Id(user.getId());
```

Comment:
find shop cart by user id, check if cartItems is not empty, create order and clear shop cart